jaiarobotics/jaiabot
# feature/echo-driver

# Summary
The echo driver will provide a way to communicate with the echo sensor (Turn on/off). The driver will query for the sensor's status every 1 second.
# Details
The approach for the echo driver is similar to the imu driver.
# Testing

1. Deploy code to a bot with a echo sensor
2. Start jaiabot services: ```sudo systemctl start jaiabot```
3. Make sure the hub is on and the services are running
4. Open jcc
5. Navigate to the JED
6. You will see a section for Echo Testing
7. Options available: Start and Stop
8. After each start or stop command sent you will need to query for a new status
9. When the echo sensor has been started it will be in the RUNNING state
10. When the echo sensor has been stopped it will be in the READY state

Another test:
1. With everything still running stop the python echo driver ```sudo systemctl stop jaiabot_echo_py.service```
2. After 20 seconds the driver should restart
3. Go through starting and stopping the echo sensor again
